### DISCUSSION TITLE
xaml中可以使用C#的`nameof()`语法么

### DISCUSSION THREAD

**Discussion by xkz1994** [BYJRK/DotNet-Discussions#280] [C# 技术交流]
xaml中可以使用C#的`nameof()`语法么

**Comment by BYJRK** [comment]
不能。这个语法类似于语法糖，编译器会把它转为实际的字符串，比如 `nameof(MyModel)` 会被转为 `"MyModel"`，而不是运行时去用反射等方式获取的

**Reply by xkz1994** [reply]
xaml中没有类似的语法糖么

**Reply by BYJRK** [reply]
可以说一说你的具体需求吗，或许你其实用不到类似这样的语法

**Reply by xkz1994** [reply]
```xaml
         <Button Content="Get"
                 Command="{Binding GetPointCommand}"
                 CommandParameter="FindPosition1" />
        <Button Content="Get"
                Command="{Binding GetPointCommand}"
                CommandParameter="FindPosition2" />
```
```csharp
    [RelayCommand]
    private async Task GetPointAsync(string name)
    {
        try
        {
            await Task.Run(() =>
            {
                var (isSuccess, result) = StageViewModel.GetStagePosition();
                if (isSuccess == false) return;

                Cache.GetType().GetProperty(name)!.SetValue(Cache, result);
            }).ConfigureAwait(false);
        }
        catch (Exception ex)
        {
            Logger.LogError(ex, "{@Name}: Get Point Failed", Name);
        }
    }
```
大佬，是这样的我这边很多按钮公用一个command 我需要区分，所以传入字符串，反射出属性设置，我想这xaml如果有nameof这种，就不用写死字符串了

**Reply by BYJRK** [reply]
没办法。你现在的方式就挺好的呀。如果想提升效率，还可以考虑提前用反射把所有属性都拿出来存到字典里，而不是每次调方法再重新反射

**Reply by xkz1994** [reply]
好的 这个替身效率确实可以哈哈 我这边这个按钮不是很常用， 用户主要是获取轴的坐标的，只需要第一次设置后面就不需要设置了。就不这么麻烦啦。谢谢大佬

**Reply by xkz1994** [reply]
大佬类似于WPF自定义CAD控件这种 重绘WPF控件的解决方案么。我这边用的`        _drawingVisual = new DrawingVisual();`的技术，我发现github上一些项目也是基于这个技术去实现重绘，但是都有同一个问题：元素过多渲染`protected override void OnRender(DrawingContext drawingContext)`就会卡。我测试了大概都是几十万没什么问题。上白万重绘元素就卡了。但是我基于GDI+WINFORM这种技术实现的自定义CAD控件，上百万不卡顿。这个DrawingVisual和GDI+绘画句柄有区别么，或者有没有更好的办法。

**Reply by xkz1994** [reply]
我说的这里面的元素：实际上就是 DrawingVisual中一个绘画对象，比如绘画矩形`_drawingContext.DrawRectangle`

**Reply by BYJRK** [reply]
你新开一个讨论吧。我不太了解自定义CAD控件
